[TF-3961] add eslint-plugin-react-hooks

joecan opened the pull request:
##### *To:*
@kevinzang @cwick @udemy/web-frontend 
##### *What:*
Adding eslint-plugin-react-hooks and corresponding rules.
https://reactjs.org/docs/hooks-rules.html
I am not sure whether 'react-hooks/exhaustive-deps' should be a warning or error.
I am also not sure whether eslint-config-udemy-react-addons was the correct scope.
##### *JIRA:*
https://udemyjira.atlassian.net/browse/TF-3961
##### *What did you test:*
none
##### *What dashboards will you be monitoring:*
none

joecan committed:
[TF-3961] add eslint-plugin-react-hooks

cwick commented on packages/eslint-config-udemy-react-addons/parts/react-hooks.js:
```diff
@@ -0,0 +1,11 @@
+        // Only call Hooks from react functions and only call Hooks at the top level
+        'react-hooks/rules-of-hooks': 'error',
+        // Verifies the list of dependencies for Hooks like useEffect and similar
+        'react-hooks/exhaustive-deps': 'warn',
```
Might as well make this `error`. Looks like messing up the dependencies will break your code in subtle ways, so better to make people deal with the issue rather than having it be a warning that can be ignored.

joecan replied:
Okay, I made it an error.

cwick reviewed (approved):
Should probably make `exhaustive-deps` an error, otherwise lgtm. Putting the hook rules in `eslint-config-udemy-react-addons` seems fine to me.

kevinzang reviewed (approved):
@joecan LGTM!

joecan committed:
[TF-3961] make react-hooks/exhaustive-deps an error

joecan committed:
Publish

 - babel-polyfill-udemy-website@9.0.18
 - babel-preset-udemy-website@11.0.15
 - eslint-config-tester@3.0.8
 - eslint-config-udemy-babel-addons@6.0.8
 - eslint-config-udemy-basics@9.0.5
 - eslint-config-udemy-jasmine-addons@8.0.9
 - eslint-config-udemy-react-addons@10.0.10
 - eslint-config-udemy-website@12.1.1
 - eslint-plugin-udemy@9.1.1
 - prettier-config-udemy-website@1.0.8
